pkg/api/v20160930/types.go

```diff
@@ -0,0 +1,194 @@
+package v20160930
+
+import (
+	neturl "net/url"
+)
+
+// ResourcePurchasePlan defines resource plan as required by ARM
+// for billing purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name,omitempty"`
+	Product       string `json:"product,omitempty"`
+	PromotionCode string `json:"promotionCode,omitempty"`
+	Publisher     string `json:"publisher,omitempty"`
+}
+
+// ContainerService complies with the ARM model of
+// resource definition in a JSON template.
+type ContainerService struct {
+	ID       string               `json:"id,omitempty"`
+	Location string               `json:"location,omitempty"`
+	Name     string               `json:"name,omitempty"`
+	Plan     ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string    `json:"tags,omitempty"`
+	Type     string               `json:"type,omitempty"`
+
+	Properties Properties `json:"properties"`
+}
+
+// Properties represents the ACS cluster definition
+type Properties struct {
+	ProvisioningState   ProvisioningState   `json:"provisioningState"`
+	OrchestratorProfile OrchestratorProfile `json:"orchestratorProfile"`
+	MasterProfile       MasterProfile       `json:"masterProfile"`
+	AgentPoolProfiles   []AgentPoolProfile  `json:"agentPoolProfiles"`
+	LinuxProfile        LinuxProfile        `json:"linuxProfile"`
+	WindowsProfile      WindowsProfile      `json:"windowsProfile"`
+	DiagnosticsProfile DiagnosticsProfile `json:"diagnosticsProfile"`
+	JumpboxProfile JumpboxProfile `json:"jumpboxProfile"`
+	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile"`
+}
+
+// ServicePrincipalProfile contains the client and secret used by the cluster for Azure Resource CRUD
+type ServicePrincipalProfile struct {
+	ClientID string `json:"servicePrincipalClientID,omitempty"`
```

Comment:
rename these to match RP, and also in Vlabs  "servicePrincipalClientID" => clientID.